cmd/gateway/s3/gateway-s3.go

```diff
@@ -198,6 +200,31 @@ var defaultAWSCredProviders = []credentials.Provider{
 	&credentials.EnvMinio{},
 }
 
+type metricsTransport struct {
+	transport *http.Transport
+	metrics   *minio.Metrics
+}
+
+func (s metricsTransport) RoundTrip(r *http.Request) (*http.Response, error) {
+	isS3Request := func() bool {
+		return !(minio.HasPrefix(r.URL.Path, minioReservedBucketPath) ||
+			minio.HasSuffix(r.URL.Path, ".js") || strings.Contains(r.URL.Path, "favicon.ico"))
```

Comment:
Can we add `strings.Contains(r.URL.Path, ".html")` to filter `http://localhost:9001/index.html/?location=`

Reply:
done